xbmc/video/VideoDatabase.cpp

```diff
@@ -4919,11 +4927,69 @@ bool CVideoDatabase::GetArtForItem(int mediaId, const MediaType &mediaType, std:
     if (nullptr == m_pDS2)
       return false; // using dataset 2 as we're likely called in loops on dataset 1
 
-    std::string sql = PrepareSQL("SELECT type,url FROM art WHERE media_id=%i AND media_type='%s'", mediaId, mediaType.c_str());
+    std::string sql;
+    if (versionId != -1 && (mediaType == MediaTypeMovie || mediaType == MediaTypeVideoVersion))
+    {
+      int versionMediaId{-1};
+      if (versionId == VIDEO_VERSION_ID_ALL)
+      {
+        //! @todo get rid of "videos with versions as folder" hack!
+        sql = PrepareSQL("SELECT videoVersionIdFile FROM movie_view WHERE idMovie = %i AND "
+                         "isDefaultVersion = 1",
+                         mediaId);
```

Comment:
"SELECT idFile FROM movie WHERE idMovie = %i" (or from movie_view) will have the same result more directly and with a lower db engine cost.

Reply:
Good catch. I will change that.